### PR TITLE
[Merged by Bors] - refactor(Algebra/Polynomial/Smeval): sort out module structures

### PR DESCRIPTION
**Pull request by loefflerd**
See discussion on Zulip here : https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Module.20structure.20diamond.3F
 
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Polynomial/Smeval.lean [hunk 1]
```diff
@@ -64,12 +64,12 @@ theorem eval_eq_smeval : p.eval r = p.smeval r := by
   rw [eval_eq_sum, smeval_eq_sum]
   rfl
 
-theorem eval₂_eq_smeval (R : Type*) [Semiring R] {S : Type*} [Semiring S] (f : R →+* S) (p : R[X])
-    (x : S) : letI : Module R S := RingHom.toModule f
-    p.eval₂ f x = p.smeval x := by
-  letI : Module R S := RingHom.toModule f
+theorem eval₂_eq_smeval (R : Type*) [Semiring R] {S : Type*} [Semiring S] [Module R S]
```
Comment (YaelDillies) [inline Mathlib/Algebra/Polynomial/Smeval.lean:67]:
```suggestion
theorem eval₂_ringHomSMulOneHom_eq_smeval (R : Type*) [Semiring R] {S : Type*} [Semiring S] [Module R S]
```
or something like that

**Comment by YaelDillies** [comment]
maintainer merge

apart from that. Thanks!

File: Mathlib/Algebra/Polynomial/Smeval.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Polynomial/Smeval.lean]: Yes thanks!

**Comment by eric-wieser** [comment]
bors merge

Thanks!

**Pull request closed without merging**